### PR TITLE
Complete algorithm to pass all tests

### PR DESCRIPTION
**Pull request by ruchiishahh**

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,62 @@
...
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //some basic if statements
+    if(request.getAttendees().isEmpty()){
+        //when there are no attendees
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    } else if(request.getDuration() > TimeRange.END_OF_DAY) {
+        //when the meeting duration is longer than a day
+        return Arrays.asList();
+    }
```
Comment (laskdaf) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:33]: you can just write a single comment at the top of the if block. something like 
"handle edge cases: if there are no attendees, then the time range is the whole day and if the request duration is longer than a day, then there's no valid time range"

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,62 @@
...
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //some basic if statements
+    if(request.getAttendees().isEmpty()){
```
Comment (laskdaf) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:27]:
```suggestion
    if(request.getAttendees().isEmpty()) {
```

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,62 @@
...
+    for(String attendee: request.getAttendees()) {
+        //traverse through all of the events occuring
+        for(Event event: events){
+            //check if that attendee is attending that event
+            if(event.getAttendees().contains(attendee)) {
+                //must block off that TimeRange
+                blockedTimes.add(event.getWhen());
+            }
+        }
+    }
```
Comment (laskdaf) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:48]: same thing here. just add a comment for the entire for block. you don't need to have one for every line, most people will understand

like what you did for `//in ascending order, add the available TimeRanges into the availableTimes list `

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (ruchiishahh) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Just fixed, thanks!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (ruchiishahh) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Fixed, thanks!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (ruchiishahh) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Fixed, thanks!

**Pull request merged**